### DISCUSSION TITLE
Full Table Scans Running on Postgres Relations Tables

### DISCUSSION THREAD

**Discussion by justinhandley** [prisma/prisma#8255] [Q&A] [answered]
On many to many relationships on Posgres databases, are the foreign keys set to be indexed, or do we have to do that manually in the database?  I couldn't figure out how to use the @@index in that scenario.  

We have implemented some many to many relationships, but our database analysis tool is telling us that full scans are being run on the actual relationship tables - the ones Prisma Autocreates between the two main data table types.  Any ideas on best practices here?

**Comment by ryands17** [comment]
@justinhandley 👋 

 We have a doc on what indexes are created in a many-to-many relationship, so this might help you out: https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#conventions-for-relation-tables-in-implicit-m-n-relations

**Reply by justinhandley** [reply]
Thanks @ryands17 - so, I have a question.  It looks like it creates an index of the unique combination of A and  B and also of B, but not of A - is there a reason for that?

We are getting a warning that our database is running full table scans on many to many relationship tables, which usually means we aren't indexing them properly.  Just trying to see if there is something we need to do to optimize that.

**Reply by janpio** [reply]
If my index knowledge is still current: A is covered by AB as well.

Can you explain a bit more what `our database analysis tool is telling us that full scans are being run on the actual relationship tables` means? Which tool? On which queries?

**Reply by sandbox-apps** [reply]
This should be easy once https://github.com/prisma/prisma/issues/8281 is implemented...

**Comment by justinhandley** [comment, marked as answer]
So, I believe that we found out what was happening.  We were getting this from Heroku reports of database table scans.  However, on deeper research, there are some instances - for example where a table doesn't have many rows - where a full table scan is more efficient than an indexed search.  The database makes the call on what it believes will be the most efficient method.  Because we are just getting our app live this week, the database is opting for full table scans over indexes, even though the indexes are there.

So, I guess that this is not an issue unless when we scale to massive table sizes the behavior continues.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
